src/core/Track.cpp

```diff
@@ -1956,7 +1956,7 @@ Track * Track::create( TrackTypes tt, TrackContainer * tc )
 		default: break;
 	}
 
-	tc->updateAfterTrackAdd();
+	tc->updateAfterTrackAdd(t);
```

Comment:
The signal/slot path seems unnecessary, specially if it involves an object from the view when the problem is in the model. Are you sure the pattern cannot be fixed on track creation?